tests/test_dataset.py

```diff
@@ -65,3 +64,25 @@ def test_context_field_is_not_null(dataset: Dataset, field: str):
 )
 def test_context_stats_field_contains_expected_keys(dataset: Dataset, field: str):
     assert field in CONTEXTS[dataset].stats
+
+
+## Batch dataset tests
+@pytest.mark.asyncio
+@pytest.mark.parametrize("dataset", BATCH_DATASETS)
+@pytest.mark.parametrize("batch_size", [1, 2, 4, 8])
+async def test_batch_size_parameter(dataset, batch_size):
+    batch_context = await dataset(batch_size=batch_size).next()
+    results = batch_context.results
+    # Check if results match expected batch size
+    assert len(results) == batch_size
```

Comment:
Maybe add a sanity check for the following:
```python
assert type(results) == list
assert all(type(result)==Context for result in result)
```

Reply:
I thought about that but I ended up thinking that typing assertion would be overkill. I will add it